### DISCUSSION TITLE
starlette < 0.17.0 doesn't give exception tracebacks when testing app

### DISCUSSION THREAD

**Discussion by kevr** [fastapi/fastapi#6433] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# asgi app definition
from fastapi import FastAPI, Request

app = FastAPI()

@app.get("/")
async def root(request: Request):
    raise Exception("Blah blah")

# pytest unit
import pytest

from fastapi.testclient import TestClient

from .asgi import app

@pytest.fixture
def client() -> TestClient:
    yield TestClient(app=app)

def test_app(client: TestClient):
    with client as request:
        response = request.get("/")
    assert response.status_code == 200
```

### Description

Run the pytest suite, an exception is raised, causing a nonsensical traceback. This nonsensical traceback is generated by `starlette==0.16.0`.

It has been resolved in `starlette==0.17.0` partially; enough that we actual get information about the original exception. FastAPI has not yet however released a version which depends on `starlette==0.17.0`, meaning we cannot yet use it through Python package managers which depend on version resolution.

If you run the aforementioned example, you will see exactly what I mean. You end up getting exceptions purely about anyio failing and not the original exception; essentially hiding the real cause of the test failure.

It currently emits a `RuntimeError("No response returned.")`

### Operating System

Linux

### Operating System Details

Arch Linux

### FastAPI Version

0.70.0

### Python Version

3.9.5

**Comment by ehiggs** [comment, marked as answer]
This will be solved by #4145

**Comment by yezz123** [comment]
Downgrade your starlette version or wait until #4145 being merged to fix this issue

**Comment by Kludex** [comment]
You can also pin the commit/branch on your requirements i.e. https://stackoverflow.com/a/13754517

For simple install, you can do: `python -m pip install git+git://github.com/simondale00/fastapi.git@upgrade-starlette-0.17.0`

But better to fork and apply your own changes rather than using third-party branches, jfyk.

**Comment by kevr** [comment]
Ahh, gotcha. Sorry, I tried searching, but guess I searched incorrectly there.

Thanks guys!
